test/es-module/test-esm-resolve-type.js

```diff
@@ -98,86 +98,154 @@ try {
+     * it will cause the resolve method to return:
+     *  {
+     *     url: '<base_path>/node_modules/my-dual-package/es/index.js',
+     *     format: 'module'
+     *  }
+     *
+     *  following testcase ensures that resolve works correctly in this case
+     *  returning the information as specified above. Source for 'url' value
+     *  is [1], source for 'format' value is [2]
+     */
+
+    const moduleName = 'my-dual-package';
 
-  const moduleName = 'my-dual-package';
-
-  const mDir = rel(`node_modules/${moduleName}`);
-  const esSubDir = rel(`node_modules/${moduleName}/es`);
-  const cjsSubDir = rel(`node_modules/${moduleName}/lib`);
-  const pkg = rel(`node_modules/${moduleName}/package.json`);
-  const esmPkg = rel(`node_modules/${moduleName}/es/package.json`);
-  const esScript = rel(`node_modules/${moduleName}/es/index.js`);
-  const cjsScript = rel(`node_modules/${moduleName}/lib/index.js`);
-
-  createDir(nmDir);
-  createDir(mDir);
-  createDir(esSubDir);
-  createDir(cjsSubDir);
-
-  const mainPkgJsonContent = {
-    type: 'commonjs',
-    main: 'lib/index.js',
-    exports: {
-      '.': {
-        'require': './lib/index.js',
-        'import': './es/index.js'
-      },
-      './package.json': './package.json',
-    }
-  };
-  const esmPkgJsonContent = {
-    type: 'module'
-  };
+    const mDir = rel(`node_modules/${moduleName}`);
+    const esSubDir = rel(`node_modules/${moduleName}/es`);
+    const cjsSubDir = rel(`node_modules/${moduleName}/lib`);
+    const pkg = rel(`node_modules/${moduleName}/package.json`);
+    const esmPkg = rel(`node_modules/${moduleName}/es/package.json`);
+    const esScript = rel(`node_modules/${moduleName}/es/index.js`);
+    const cjsScript = rel(`node_modules/${moduleName}/lib/index.js`);
 
-  fs.writeFileSync(pkg, JSON.stringify(mainPkgJsonContent));
-  fs.writeFileSync(esmPkg, JSON.stringify(esmPkgJsonContent));
-  fs.writeFileSync(esScript,
-                   'export function esm-resolve-tester() {return 42}');
-  fs.writeFileSync(cjsScript,
-                   `module.exports = { 
-                     esm-resolve-tester: () => {return 42}}`
-  );
+    createDir(nmDir);
+    createDir(mDir);
+    createDir(esSubDir);
+    createDir(cjsSubDir);
+
+    const mainPkgJsonContent = {
+      type: 'commonjs',
+      main: 'lib/index.js',
+      exports: {
+        '.': {
+          'require': './lib/index.js',
+          'import': './es/index.js'
+        },
+        './package.json': './package.json',
+      }
+    };
+    const esmPkgJsonContent = {
+      type: 'module'
+    };
+
+    fs.writeFileSync(pkg, JSON.stringify(mainPkgJsonContent));
+    fs.writeFileSync(esmPkg, JSON.stringify(esmPkgJsonContent));
+    fs.writeFileSync(esScript,
+                     'export function esm-resolve-tester() {return 42}');
+    fs.writeFileSync(cjsScript,
+                     `module.exports = { 
+                        esm-resolve-tester: () => {return 42}}`
+    );
```

Comment:
For other devs who come after, please fix these awkward and difficult-to-read line-breaks and indentation (which may have come from the horrible autofixer).

Reply:
I tried to format this a bit differently. Pls let me know if it's better. Indeed I did that because of lint errors and mainly because of the max line length of 80. Hope it is better now.